src/cmd-buildextend-installer

```diff
@@ -440,41 +428,39 @@ def generate_iso():
             isofh.write(struct.pack(fmt, b'coreiso+', offset, initrd_ignition_padding))
             print(f'Embedded {initrd_ignition_padding} bytes Ignition config space at {offset}')
 
-    kernel_name = f'{base_name}-{args.build}-{image_type}-kernel-{basearch}'
-    initramfs_name = f'{base_name}-{args.build}-{image_type}-initramfs.{basearch}.img'
-    kernel_file = os.path.join(builddir, kernel_name)
-    initramfs_file = os.path.join(builddir, initramfs_name)
-    shutil.copyfile(os.path.join(tmpisoimages, kernel_img), kernel_file)
-    with open(initramfs, 'rb') as fsrc:
-        with open(initramfs_file, 'wb') as fdst:
-            shutil.copyfileobj(fsrc, fdst)
-            if is_live:
-                # Verify ISO initrd padding and truncate it away
-                pad_offset = fsrc.tell() - initrd_ignition_padding
-                fsrc.seek(pad_offset)
-                if fsrc.read() != bytes(initrd_ignition_padding):
-                    raise Exception(f"Expected {initrd_ignition_padding} bytes of trailing zeroes in initrd, didn't find it")
-                fdst.truncate(pad_offset)
-
-    kernel_checksum = sha256sum_file(kernel_file)
-    initramfs_checksum = sha256sum_file(initramfs_file)
-    checksum = sha256sum_file(tmpisofile)
-
     buildmeta['images'].update({
         meta_keys['iso']: {
             'path': iso_name,
-            'sha256': checksum
-        },
-        meta_keys['kernel']: {
-            'path': kernel_name,
-            'sha256': kernel_checksum
-        },
-        meta_keys['initramfs']: {
-            'path': initramfs_name,
-            'sha256': initramfs_checksum
+            'sha256': sha256sum_file(tmpisofile)
         }
     })
     os.rename(tmpisofile, f"{builddir}/{iso_name}")
+
+    if not args.no_pxe:
+        kernel_name = f'{base_name}-{args.build}-{image_type}-kernel-{basearch}'
+        initramfs_name = f'{base_name}-{args.build}-{image_type}-initramfs.{basearch}.img'
+        kernel_file = os.path.join(builddir, kernel_name)
+        initramfs_file = os.path.join(builddir, initramfs_name)
+        shutil.copyfile(os.path.join(tmpisoimages, kernel_img), kernel_file)
+        cp_reflink(base_initramfs, initramfs_file)
+        # Append the rootfs squashfs to the PXE initramfs, so that it can be used
+        # directly for installs.
+        if is_live:
+            with tempfile.TemporaryDirectory(prefix='initramfs', dir=tmpdir) as tmproot:
+                shutil.move(tmp_squashfs, tmproot)
+                # Compression is redundant but the kernel requires it
+                extend_initrd(initramfs_file, tmproot, compression=False)
```

Comment:
Hmm, don't we also need to extend with the osmet CPIO here? Otherwise I think this breaks osmet installs in the PXE path (we sadly don't have CI for that path yet).